lldb/source/Plugins/SymbolFile/DWARF/DWARFDebugInfoEntry.cpp

```diff
@@ -1230,6 +1230,29 @@ DWARFDebugInfoEntry::GetAbbreviationDeclarationPtr(const DWARFUnit *cu) const {
   return nullptr;
 }
 
+bool DWARFDebugInfoEntry::IsGlobalOrStaticVariable() const {
+  if (Tag() != DW_TAG_variable)
+    return false;
+  const DWARFDebugInfoEntry *parent_die = GetParent();
+  while (parent_die != nullptr) {
+    switch (parent_die->Tag()) {
+    case DW_TAG_subprogram:
```

Comment:
This is counter-intuitive. How is
```
void f() {
static int g_i = 0;
}
```
represented in DWARF?

Reply:
Can you please comment on the upstream version? This is a cherry-pick

Reply:
Done (https://reviews.llvm.org/D77698)